packages/orbit-components/src/PricingTable/__tests__/index.test.js

```diff
@@ -1,188 +1,108 @@
 // @flow
 
 import * as React from "react";
-import { mount } from "enzyme";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
 
+import useMediaQuery from "../../hooks/useMediaQuery";
 import Button from "../../Button";
 import Badge from "../../Badge";
 import PricingTableItem from "../PricingTableItem";
 import PricingTable from "../index";
-import PricingTableContext from "../PricingTableContext";
-
-const onClick = jest.fn();
-const onClick2 = jest.fn();
-const dataTest = "test";
-const name = "name";
-const price = "$749";
-const actions = <Button>Buy</Button>;
-const content = <ul>List</ul>;
-const image = <img alt="" />;
-const badge = <Badge>Badge</Badge>;
-
-jest.mock("../../hooks/useMediaQuery", () => {
-  return () => {
-    return {
-      isTablet: false,
-    };
-  };
+
+jest.mock("../../hooks/useMediaQuery", () => jest.fn());
+const useMediaQueryMock: JestMockFn<[], $Shape<$Call<typeof useMediaQuery>>> = useMediaQuery;
+
+afterEach(() => {
+  useMediaQueryMock.mockRestore();
 });
 
-describe("PricingTableItem mobile", () => {
-  const component = mount(
-    <PricingTableContext.Provider value={{ basis: 0 }}>
-      <PricingTableItem
-        dataTest={dataTest}
-        name={name}
-        price={price}
-        featureIcon={image}
-        badge={badge}
-        action={actions}
-        active
-        compact
-        onClick={onClick}
-      >
-        {content}
-      </PricingTableItem>
-    </PricingTableContext.Provider>,
-  );
-
-  const pricingTableItem = component.find("PricingTableItem__StyledPricingTableItem");
-  it("shouldn't render children", () => {
-    expect(component.find("ul").exists()).toBe(false);
-  });
-  it("should have name", () => {
-    expect(pricingTableItem.find("Text").at(0).children().text()).toBe(name);
-  });
-  it("should render data-test", () => {
-    expect(pricingTableItem.render().prop("data-test")).toBe(dataTest);
-  });
-  it("should have price", () => {
-    expect(pricingTableItem.find("Text").at(1).children().text()).toBe(price);
-  });
-  it("should have featureIcon", () => {
-    expect(pricingTableItem.find("img").exists()).toBe(true);
-  });
-  it("should have badge", () => {
-    expect(pricingTableItem.find("PricingTableItem__StyledBadgeWrapperContent").exists()).toBe(
-      true,
+describe("PricingTableItem", () => {
+  it("should have expected DOM output", () => {
+    useMediaQueryMock.mockImplementation(() => ({ isTablet: false }));
```

Comment:
sorry, i did not understand what you mean, it checks with `mobileDescription` prop and that needs to be separated from the first one and also be lower than `isTablet`

Reply:
This test passes even if `isTablet` is `true` because the component only checks for `isDesktop`, which is `undefined` in this test, which is falsy so the test passes, but not because of the reason that this mock implies.